src/style.rs

```diff
@@ -393,16 +386,16 @@ pub struct Style {
     /// How should items be aligned relative to the main axis?
     pub justify_content: JustifyContent,
     /// How should the position of this element be tweaked relative to the layout defined?
-    pub position: Rect<Dimension>,
+    pub position: Rect<LengthPercentageAuto>,
     /// How large should the margin be on each side?
-    pub margin: Rect<Dimension>,
+    pub margin: Rect<LengthPercentageAuto>,
     /// How large should the padding be on each side?
-    pub padding: Rect<Dimension>,
+    pub padding: Rect<LengthPercentage>,
     /// How large should the border be on each side?
-    pub border: Rect<Dimension>,
+    pub border: Rect<LengthPercentage>,
     // Gap
     /// How large should the gaps between items in a grid or flex container be?
-    pub gap: Size<Dimension>,
+    pub gap: Size<LengthPercentage>,
```

Comment:
Do you mean like a sub-enum that takes a definition from a master one with all the variants. Unfortunately that's not yet a Rust feature, although it may be one day (it's been proposed before).

Reply:
Alright, good to know :)